### PR TITLE
feat: refactor runtime plugin

### PR DESCRIPTION
**Pull request by caohuilin**
## Summary

- add `modern.runtime.ts` file to add runtime config 
- hoc hook support config params to support router plugin custom basename
- remove provider hook for runtime plugin
- refactor runtime plugin runner
## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/runtime/plugin-runtime/src/core/plugin/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export function registerPlugin(
+  internalPlugins: Plugin[],
+  runtimeConfig?: RuntimeConfig,
+  customRuntime?: typeof runtime,
+) {
+  const { plugins = [] } = runtimeConfig || {};
+  (customRuntime || runtime).usePlugin(...internalPlugins, ...plugins);
+  const runner = (customRuntime || runtime).init();
+  // It is necessary to execute init after usePlugin, so that the plugin can be registered successfully.
+  setGlobalRunner(runner);
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/runtime/plugin-runtime/src/core/plugin/index.ts:19]: The runner‘s initial needs  to be after the `usePlugin` function. In the new design, there doesn't seem to be a place to put the logic for `setGlobalRunner`.

File: packages/runtime/plugin-runtime/src/core/compatible.tsx [hunk 2]
```diff
@@ -27,30 +32,16 @@ function isClientArgs(
...
-  loaderManager: createLoaderManager({}),
-  runner,
-  isBrowser: true,
-  routeManifest:
-    typeof window !== 'undefined' && (window as any)[ROUTE_MANIFEST],
-});
-
 export const createApp = ({
   plugins,
+  runtime,
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/runtime/plugin-runtime/src/core/compatible.tsx:37]: Yes. In the test cases, we need to execute multiple `createApp` instances simultaneously, and they must not share the runtime.

File: packages/runtime/plugin-runtime/src/core/compatible.tsx [hunk 3]
```diff
@@ -64,22 +55,14 @@ export const createApp = ({
...
-      );
     };
 
     if (App) {
       hoistNonReactStatics(WrapperComponent, App);
     }
 
     const HOCApp = runner.hoc(
-      { App: WrapperComponent },
+      { App: WrapperComponent, config: globalProps || {} },
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/runtime/plugin-runtime/src/core/compatible.tsx:65]: In the garfish, the basename is passed through globalProps

File: packages/runtime/plugin-runtime/src/core/compatible.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zllkjc) [reply packages/runtime/plugin-runtime/src/core/compatible.tsx:44]: OK, maybe we can add a comment~

**Review by zllkjc — APPROVED** [review]

**Pull request merged**